### PR TITLE
Add image cache for employee avatars

### PR DESCRIPTION
**Pull request by gfontenot**
This adds a simple in-memory image cache to reduce our network calls when
displaying employees. Each time the app is launched, it fetches each avatar
only once, pulling them from the local cache after that.

I was playing around with code organization here. Would love to get some
feedback on that.

### CODE REVIEW

File: roster/Classes/Models/ImageCache.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+
+    func didRecieveMemoryWarning() {
+        removeAllObjects()
+    }
+}
+
+extension ImageCache {
```
Comment (hyperspacemark) [inline roster/Classes/Models/ImageCache.swift]: What does this public extension get us?

**Comment by hyperspacemark** [comment]
Just one question. Otherwise good to go. :+1:

File: roster/Classes/Models/ImageCache.swift (hunk 1 above)
Comment (gfontenot) [inline roster/Classes/Models/ImageCache.swift]: Code seperation? I'm not entirely sure. I was just playing around with the concept of moving the class methods into an extension. I'm not sure if I like it, just playing around with some ideas.

Comment (hyperspacemark) [inline roster/Classes/Models/ImageCache.swift]: I like the private extension for the singleton instance accessor. Not sold on this one.

**Comment by tonyd256** [comment]
:+1:

**Pull request merged**